Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -271,8 +271,8 @@ static bool IsInside(in List<List<Vector2>> polygons, in Vector2 p)
                     Vector2 p1 = polygon[index];
                     Vector2 p2 = polygon[(index + 1) % polygon.Count];
                     int intersectionType = UnityVectorExtensions.FindIntersection(p, camRayEndFromCamPos2D, p1, p2, 
-                        out _);
-                    if (intersectionType == 2)
+                        out var intersection);
+                    if (intersectionType == 2 && (intersection - p2).sqrMagnitude < 0.01f)
```

Comment:
There is no need to check for (intersection - p2).sqrMagnitude < 0.01f here. 
I checked and for me the camera is never Inside with this extra condition. If I remove it, then it correctly identifies inside/outside.